### PR TITLE
Cleanup and document DataStorm Slice definitions

### PR DESCRIPTION
**Pull request by pepone**
This PR reformats the DataStorm Slice definitions and adds Doc comments. I didn't document all the interfaces yet.

### CODE REVIEW

File: slice/DataStorm/Sample.ice [hunk 1]
```diff
@@ -5,34 +5,25 @@
 
 module DataStorm
 {
+    /// The sample event matches the operation used by the DataWriter to update
```
Comment (bernardnormier) [inline slice/DataStorm/Sample.ice:8]: Use 120 columns for Slice files.

File: cpp/src/DataStorm/Contract.ice [hunk 2]
```diff
@@ -4,244 +4,284 @@
...
 module DataStormContract
 {
+    enum ClearHistoryPolicy
+    {
+        OnAdd,
+        OnRemove,
+        OnAll,
+        OnAllExceptPartialUpdate,
+        Never
+    };
```
Comment (bernardnormier) [inline cpp/src/DataStorm/Contract.ice:21]: no semicolon

File: cpp/src/DataStorm/Contract.ice [hunk 3]
```diff
@@ -4,244 +4,284 @@
...
+
+    /// A dictionary of <long, long>
```
Comment (bernardnormier) [inline cpp/src/DataStorm/Contract.ice:23]: That's not a good doc-comment.

**Review by bernardnormier — APPROVED** [review]

**Review by externl — APPROVED** [review]

File: cpp/src/DataStorm/Contract.ice (hunk 3 above)
Comment (pepone) [reply cpp/src/DataStorm/Contract.ice]: Removed, I would add more comments in a follow-up PR.

**Pull request merged**